source/brailleDisplayDrivers/hidBrailleStandard.py

```diff
@@ -103,16 +104,27 @@ def __init__(self, port="auto"):
 				continue  # Couldn't connect.
 			if self._dev.usagePage != HID_USAGE_PAGE_BRAILLE:
 				log.debug("Not braille")
+				self._dev.close()
 				continue
 			cellValueCaps = self._findCellValueCaps()
-			if cellValueCaps:
+			if len(cellValueCaps) > 0:
+				if any(x.ReportCount != cellValueCaps[0].ReportCount for x in cellValueCaps):
+					log.warning("Found multi-line display with an irregular shape, ignoring.")
+					self._dev.close()
+					continue
+				self.numRows = len(cellValueCaps)
+				self.numCols = cellValueCaps[0].ReportCount
+				self._maxNumberOfCells = self.numCells
 				self._cellValueCaps = cellValueCaps
-				self._numberOfCellsValueCaps = self._findNumberOfCellsValueCaps()
-				self.numCells = self._maxNumberOfCells = cellValueCaps.ReportCount
+				if self.numRows == 1:
+					self._numberOfCellsValueCaps = self._findNumberOfCellsValueCaps()
+				elif self._findNumberOfCellsValueCaps():
+					log.warning("The number of braille cells usage is not supported on multi-line displays")
```

Comment:
Is this still accurate as an alternative description?

```suggestion
					log.warning("Reserved braille cells usage is not supported on multi-line displays")
```